### ISSUE TITLE
RFC: commit structure methodology

### ISSUE THREAD

**Issue by bhaibel** [wecohere/nourish.party#42]
I'm noticing a PR structure pattern where there's incidental infrastructure pulled in (`decent_exposure`, rspec feature tests) alongside actual feature work, and the entire PR is then squash-merged onto master. I am uncomfortable with this pattern for the following reasons:

- large squash commits on master make it difficult to track the historical reasons for decisions.
- it's hard to have targeted PR discussions -- I have qualms about both `decent_exposure` and rspec feature tests, but if I raise those concerns on a feature PR then all of a sudden I'm blocking the feature. I don't like that social dynamic. It creates a situation in which people can barrel over other people with "finished code wins" and discourages thoughtfulness.

I think that there may be an assumption that this is temporary because the project is new, and that it will settle down as the project matures. In my experience, that does not happen naturally, it happens because people choose to be intentional about PR size and scope norms. I would like us to set strong norms around patch size and scope early. My preferred norms -

- Small PRs always. If a PR exceeds *n1* files changed or *n2* lines changed, let's take it as a sign that it introduces too much to be adequately reviewed and look at ways to split the story.
- Separate PRs for infrastructure decisions, possibly accompanied by RFC issues. If it comes up that a story requires new infrastructure to be viable, then that story takes multiple PRs. Similarly, if a story requires a refactor to be viable, then that story takes multiple PRs and the refactor comes first.
- Continue using squash merges.

I'm also open to alternate processes for reducing the size of commits on master, like rebase-focused processes, but those only address my commit size concerns. I'd like us to find a solution that also addresses my comms structure concerns.

TLDR: I think this project will be best served if we prefer thoughtfulness, historical record of decisionmaking, and transparency of values over velocity, and I see our current PR strategy as not doing that.
